### ISSUE TITLE
Ploidy estimation difference

### ISSUE THREAD

**Issue by xtmgah** [mskcc/facets#64]
Hello:

I am running the facets to a WGS samples. However, I found when I try the different parameters, I got different ploidy estimation. the purity predictions were pretty similar. Can you let me know which result should I more trust? Thanks so much. 

First parameter estimated the ploidy is 3.17

#Test preprosample normal read filter ndepth 15
#Test preprosample cval 75
#Test preprosample window size snp.nbhd 5000
#Test prosample cval: 500

 second parameter estimated the ploidy is 1.62

#Test preprosample normal read filter ndepth 15
#Test preprosample cval 75
#Test preprosample window size snp.nbhd 5000
#Test prosample cval: 500

[Ploidy issue.pdf](https://github.com/mskcc/facets/files/1515359/Ploidy.issue.pdf)

**Comment by veseshan** [comment]
This may be the case of identifiability that cannot be resolved from DNAseq data and would require alternate karyotyping approach. A cancer genome with genome doubling i,e chromosomes with copy numbers (2,0), (3,1) and (4,2) present clonally is indistinguishable from one with sub clonal and subclonal losses at (1,0) if the subclonal proportion is just about halfway point. The two solutions seem to be the manifestation of that. If you can provide fit$cncf data from the two runs we can puzzle over it more.

Venkat

**Comment by xtmgah** [comment]
Hey Venkat:

Thanks for your quickly response. Here is the cncf data for both tests. Let me know what's your suggestion. 
[IGC-03-1254-T02_15_75_5000_500.cncf.txt](https://github.com/mskcc/facets/files/1515436/IGC-03-1254-T02_15_75_5000_500.cncf.txt)
[IGC-03-1254-T02_15_75_10000_1000.cncf.txt](https://github.com/mskcc/facets/files/1515437/IGC-03-1254-T02_15_75_10000_1000.cncf.txt)

**Comment by veseshan** [comment]
Can you also send the naive fits in out$out from the two runs? Thanks.

**Comment by veseshan** [comment]
By the way are you using emcncf2? (There is a clonal.cluster column in fit$cncf). Unfortunately that program has some issues which is we print a warning message when it is invoked.

**Comment by xtmgah** [comment]
Yes. We used emcnf2. So, Do you think this will cause the ploidy issue? Please check the following link to download the RData file (including the oo and fit), which can be load to R. here is the core code for this: 

input <- paste0("Result/",sample,".pileup.txt.gz")
rcmat=readSnpMatrix(input)
xx <- preProcSample(rcmat,ndepth = ndepth,cval = cval,snp.nbhd = snp.nbhd)
oo <- procSample(xx,cval=cval2)
fit <- emcncf2(oo)


https://www.dropbox.com/sh/8xg01gnu7b7once/AAAtA_DqzX0RqieTo18I3vqBa?dl=0

**Comment by xtmgah** [comment]
I just try the "emcncf", and both result are consistent, the ploidy is predict around 1.6. So, it should be caused by "emcncf2" function. So, we should replace all the prediction from "emcncf2" to "emcncf" ?? Thanks.

**Comment by veseshan** [comment]
You are right that emcncf and emcncf2 results are typically similar. However since emcncf has more flexibility cf, tcn and lcn could differ. I will download the RData files ASAP.

Thanks

**Comment by xtmgah** [comment]
Hello veseshan:

Did you have any chance look at the detail? can you let me know why there are unstable to predict the ploidy in this sample? Thanks so much.

**Comment by veseshan** [comment]
Haven't had a chance to look at it. Will do asap.

**Comment by veseshan** [comment]
The data files gave me a better understanding of what's being done. The top figure comes from snp.nbhd=5000 & cval=500 and the bottom from snp.nbhd=10000 & cval=1000. [The annotation in the figure says 5000, 500 for both and I thought they were two different runs of the same data.] Even though this is WGS data snp.nbhd of 10000 will lose most of the information since it will set the average difference between consecutive loci in the analysis to be 10000. So snp.nbhd of 5000 is the better choice. In since typical insert size is ~500 you can even go lower to say 2000 and still not have any serial correlation issue. However because of the density of the data and the prevalence of hets you should use a large cval (1000 or even 2000) to avoid some artifacts. There are clear signs of genome duplication. But there may be some emcncf2 issue in terms of copy number assignment. Try snp.nbhd=2000 and cval=2000 with emcncf.

**Comment by xtmgah** [comment]
Great. Thanks, Will try this new parameters now and will let you know later. Do you know other method or software can be used to identify if the genome doubling or not using WGS data? Just want to see if it can be verified by other  method.

**Comment by veseshan** [comment]
Yes. Karyotyping or FISH. For this sample if you have FISH probes for centromeres of 2, 5 and 7 you should see 4 copies of 2 and 7 and only 2 of 5. If on the other hand you only saw 2,2 and 1 then it is not genome doubled.

**Comment by xtmgah** [comment]
![igc-03-1254-t02_15_75_2000_2000 cncf](https://user-images.githubusercontent.com/4236148/33465410-bde115a8-d615-11e7-8f52-16308e701536.png)


Please check the result using 2000,2000. The ploid here is 1.6.... look's like no genome doubling??

**Comment by veseshan** [comment]
Can you make the oo object available? I'm puzzled.

**Comment by xtmgah** [comment]
Here is the oo object RData. Thanks for checking this. I am try to run 2000,500, and will send you later

https://www.dropbox.com/s/21zugbhun7f50qq/IGC-03-1254-T02_15_75_2000_2000.RData?dl=0

**Comment by xtmgah** [comment]
![igc-03-1254-t02_15_75_2000_500 cncf](https://user-images.githubusercontent.com/4236148/33467440-83ad7e00-d622-11e7-8eb9-cc5fe5b46d4c.png)


Here is the result for run 2000, 500, Now, I am confused. Also, please check the oo RData in the following link:

https://www.dropbox.com/s/biqiolpabo8mk0q/IGC-03-1254-T02_15_75_2000_500.RData?dl=0

**Comment by veseshan** [comment]
The problem arose because cval=2000 is too large and clobbered the signal in chromosome 19. The segmentation result for 19 is:

```
> round(oo$out[oo$out$chrom==19,], digits=3)
   chrom seg num.mark  nhet cnlr.median  mafR segclust cnlr.median.clust mafR.clust    cf tcn lcn
46    19  46    26530 14900       0.012 0.021       18             0.012      0.021 0.154   0   0
```
whereas for cval=500 it is

```
> round(oo$out[oo$out$chrom==19,], digits=3)
   chrom seg num.mark  nhet cnlr.median  mafR segclust cnlr.median.clust mafR.clust    cf tcn lcn
63    19  63       20     0       0.424 0.000       32             0.397         NA 0.734   4  NA
64    19  64     8690  4859      -0.114 0.060       12            -0.156      0.060 0.445   3   1
65    19  65    17818 10040       0.076 0.001       24             0.091      0.008 0.529   4   2
```
So 19p and 19q have different copy number states and the large cval lost it. Note also that in cval=2000 the entire chromosome is called as homozygous deletion. This can't happen. So there is genome doubling in the sample.

Take home message don't make cval too large - 500 seems large enough.

**Comment by xtmgah** [comment]
Great. Thank for explain this. I will try the new parameters (2000,500) to other samples and see if I still have similar issue. Thanks for your help.
